src/v2/components/AttachmentPicker/AttachmentPicker.tsx

```diff
@@ -0,0 +1,309 @@
+import React, { useEffect, useState } from 'react';
+import {
+  AppState,
+  AppStateStatus,
+  ImageBackground,
+  Keyboard,
+  Platform,
+  StyleSheet,
+  View,
+} from 'react-native';
+import BottomSheet, {
+  BottomSheetFlatList,
+  BottomSheetHandleProps,
+  TouchableOpacity,
+} from '@gorhom/bottom-sheet';
+
+import { useAttachmentPickerContext } from '../../contexts/attachmentPickerContext/AttachmentPickerContext';
+import { useTheme } from '../../contexts/themeContext/ThemeContext';
+import { getPhotos } from '../../native';
+import { vh, vw } from '../../utils/utils';
+
+import type { AttachmentPickerErrorProps } from './components/AttachmentPickerError';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: '#FFFFFF',
+    flexGrow: 1,
+  },
+  overlay: {
+    alignItems: 'flex-end',
+    backgroundColor: '#00000080',
+    flex: 1,
+  },
+});
+
+const screenHeight = vh(100);
+
+type AttachmentImageProps = {
+  ImageOverlaySelectedComponent: React.ComponentType;
+  onPress: () => void;
+  selected: boolean;
+  uri: string;
+  numberOfAttachmentPickerImageColumns?: number;
+};
+
+const AttachmentImage: React.FC<AttachmentImageProps> = (props) => {
+  const {
+    ImageOverlaySelectedComponent,
+    numberOfAttachmentPickerImageColumns,
+    onPress,
+    selected,
+    uri,
+  } = props;
+  const {
+    theme: {
+      attachmentPicker: { image, imageOverlay },
+    },
+  } = useTheme();
+
+  const size = vw(100) / (numberOfAttachmentPickerImageColumns ?? 3) - 2;
+
+  return (
+    <TouchableOpacity onPress={onPress}>
+      <ImageBackground
+        source={{ uri }}
+        style={[
+          {
+            height: size,
+            margin: 1,
```

Comment:
lol `margin: 1`?

Reply:
Haha yeah it ends up being 2 cause it's around all of the adjacent images...can modify it pretty well too.